docs/features/incremental-generators.cookbook.md

```diff
@@ -0,0 +1,810 @@
+            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs("UserBool"));
+        }
+    }
+
+    public int Count
+    {
+        get => _intProp;
+        set
+        {
+            _intProp = value;
+            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs("Count"));
+        }
+    }
+
+    public event PropertyChangedEventHandler PropertyChanged;
+}
+
+```
+
+### Package a generator as a NuGet package
+
+**User scenario**: As a generator author I want to package my generator as a NuGet package for consumption.
+
+**Solution:** Generators can be packaged using the same method as an Analyzer would.
+Ensure the generator is placed in the `analyzers\dotnet\cs` folder of the package for it to be automatically added to the users project on install.
+
+For example, to turn your generator project into a NuGet package at build, add the following to your project file:
+
+```xml
+  <PropertyGroup>
+    <GeneratePackageOnBuild>true</GeneratePackageOnBuild> <!-- Generates a package at build -->
+    <IncludeBuildOutput>false</IncludeBuildOutput> <!-- Do not include the generator as a lib dependency -->
+  </PropertyGroup>
+
+  <ItemGroup>
+    <!-- Package the generator in the analyzer directory of the nuget package -->
+    <None Include="$(OutputPath)\$(AssemblyName).dll" Pack="true" PackagePath="analyzers/dotnet/cs" Visible="false" />
+  </ItemGroup>
+```
+
+### Use functionality from NuGet packages
+
+**User Scenario:** As a generator author I want to rely on functionality provided in NuGet packages inside my generator.
+
+**Solution:** It is possible to depend on NuGet packages inside of a generator, but special consideration has to be taken for distribution.
+
+Any *runtime* dependencies, that is, code that the end users program will need to rely on, can simply be added as a dependency of the generator NuGet package via the usual referencing mechanism.
+
+For example, consider a generator that creates code that relies on `Newtonsoft.Json`. The generator does not directly use the dependency, it just emits code that relies on the library being referenced in the users compilation. The author would add a reference to `Newtonsoft.Json` as a public dependency, and when the user adds the generator package it will referenced automatically.
+
+```xml
+<Project>
+  <PropertyGroup>
+    <GeneratePackageOnBuild>true</GeneratePackageOnBuild> <!-- Generates a package at build -->
+    <IncludeBuildOutput>false</IncludeBuildOutput> <!-- Do not include the generator as a lib dependency -->
+  </PropertyGroup>
+
+  <ItemGroup>
+    <!-- Take a public dependency on Json.Net. Consumers of this generator will get a reference to this package -->
+    <PackageReference Include="Newtonsoft.Json" Version="12.0.1" />
+
+    <!-- Package the generator in the analyzer directory of the nuget package -->
+    <None Include="$(OutputPath)\$(AssemblyName).dll" Pack="true" PackagePath="analyzers/dotnet/cs" Visible="false" />
```

Comment:
I know this is coppied from the cookbook, but should we consider changing this to use System.Text.Json?

Reply:
I don't think it particularly matters tbh.